### PR TITLE
Run unadded additional dashcard series using card query endpoint

### LINKED ISSUE #20637: Can't combine saved questions on a dashboard card
Trying to add e.g. an "Orders per month" saved question to a dashboard card like "Count of people per month" incorrectly tells me that the question I'm trying to add is incompatible.

I encountered this both on our Stats instance as well as on a local instance running master.

Calling this a P2 until we can verify if this is only a problem with H2 or if it's more widespread.

image

Repro:
https://www.loom.com/share/a0d9606a8a9a486fbe15cf4738514822

### PR DESCRIPTION
**Pull request by daltojohnso**
Resolves #20637 
Reproduces #20637

Similar to newly added dashcards preceding a dashboard save we need to run the query for the card newly added from the `AddSeriesModal` using the `/api/card/:cardId/query` endpoint instead of the dashcard endpoint. I've added `isNewAdditionalSeriesCard` to determine whether a card is a newly added additional series and I am using it in the `fetchCardData` action.

**Testing**
Follow steps in #20637 

https://user-images.githubusercontent.com/13057258/155388355-bd142c5c-e7a1-44f8-adb1-ab906f4a20da.mov

### CODE REVIEW

**Review by flamber — APPROVED** [review]
Works with every combination I throw at it

File: frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    // make sure the card query endpoint was used
+    cy.wait("@additionalSeriesCardQuery");
+
+    cy.get(".AddSeriesModal").within(() => {
+      cy.findByText("Done").click();
+    });
+    cy.findByText("Save").click();
+
+    // refresh the page and make sure the dashcard query endpoint was used
+    cy.reload();
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js:32]: It is slightly concerning that the reload happens faster than the `GET /api/dashboard/2` so it cancels that request.
image

Just to be on the safe side, maybe it's worth waiting for all requests before reloading. Something like this:
```js
function saveDashboard() {
  cy.intercept("PUT", "/api/dashboard/*").as("updateDashboard");
  cy.intercept("PUT", "/api/dashboard/*/cards").as("updateDashCards");
  cy.intercept("GET", "/api/dashboard/*").as("loadDashboard");

  cy.findByText("Save").click();

  cy.wait(["@updateDashboard", "@updateDashCards", "@loadDashboard"]);
}
```

If I do that locally, there are no cancelled requests any more:
image

File: frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    cy.wait("@dashcardQuery");
+    cy.wait("@additionalSeriesDashcardQuery");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js:34]:
```suggestion
    cy.wait(["@dashcardQuery", "@additionalSeriesDashcardQuery"]);
```
```suggestion
    cy.wait("@dashcardQuery");
    cy.wait("@additionalSeriesDashcardQuery");
```

File: frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js (hunk 1 above)
Comment (daltojohnso) [reply frontend/test/metabase/scenarios/dashboard/reproductions/20637-add-series-to-dashcard.cy.spec.js:32]: yeah, seems like a good idea to add

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**